flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/functions/aggfunctions/DeclarativeAggregateFunction.java

```diff
@@ -62,7 +62,7 @@
 	/**
 	 * All types of the aggregate buffer.
 	 */
-	public abstract InternalType[] getAggBufferTypes();
+	public abstract LogicalType[] getAggBufferTypes();
```

Comment:
Regarding to `DeclarativeAggregateFunction`, I think it is an API expose to "users" or out of table system. Maybe we should use `DataType` for `getAggBufferTypes` and `getResultType`.

Reply:
Yes, DataType should be used at the edges.

Reply:
I have different thought.
The `DeclarativeAggregateFunction` does not expose physical representation. It is the interface to description logic computing for user.

>The DataType class has two responsibilities: declaring a logical type and giving hints about the physical representation of data to the optimizer. While the logical type is mandatory, hints are optional but useful at the edges to other APIs.

So I think we should use `LogicalType` for `getAggBufferTypes` and `getResultType`.

Reply:
The user should not be confronted to two different type stacks. That's why the description in `DataType` calls it "hints", a user should always use DataType. If he adds a hint where it is not appropriate, we just don't need to use those hints.

Reply:
Yes, I got it, it is better to let user just know one single type stack.